ipa-core/src/protocol/dp/mod.rs

```diff
@@ -236,15 +242,16 @@ where
                 return Err(EpsilonOutOfBounds);
             }
 
-            let per_user_credit_cap = 2_f64.powi(i32::try_from(SS_BITS).unwrap());
+            let per_user_credit_cap = 2_u32.pow(u32::try_from(SS_BITS).unwrap());
```

Comment:
suppose we are setting `per_user_credit_cap` to, say, 5, but the `SS_BITS` are larger (for 5, you'd need at least 3 bits). doesn't this result in more noise than needed? more generally, why infer this from `SS_BITS` when we have a more precise value as an input to the query?

Reply:
Right now the code only supports a `per_user_credit_cap` that is a power of two and when calling `oprf_ipa` the conversions happens to pass in a generic parameters of the power of 2 `SS_BITS`  https://github.com/private-attribution/ipa/blob/77ac0b009afcdef2152ce5b9ed0499f4f7cdebb5/ipa-core/src/query/runner/oprf_ipa.rs#L123 

By guess is that it is capping that most heavily uses this power of 2 assumption.  When we shift to the hybrid without a capping stage is probably a good time to refactor to support a more flexible value for the sensitivity in the query. 

It will add more noise than needed if the true cap is less than a power of two, but the caller of the API get around this by scaling up their inputs (by (`next_power_of_two/true_cap`)) and then rescale the outputs.